pull/github.go

```diff
@@ -356,20 +380,36 @@ func (ghc *GitHubContext) loadCollaboratorData() error {
 					Permission string  `graphql:"permission"`
 					Node       v4Actor `graphql:"node"`
 				}
-			} `graphql:"collaborators(first: 100, after: $collaboratorCursor)"`
+			} `graphql:"collaborators(first: 100, after: $collaboratorCursor, affiliation: $affiliation)"`
 		} `graphql:"repository(owner: $owner, name: $name)"`
 	}
+	affiliation := githubv4.CollaboratorAffiliationAll
+	if direct {
+		affiliation = githubv4.CollaboratorAffiliationDirect
+	}
+
+	if direct {
+		if ghc.directCollaborators != nil {
+			return ghc.directCollaborators, nil
+		}
+	} else {
+		if ghc.collaborators != nil {
+			return ghc.collaborators, nil
+		}
+	}
```

Comment:
Instead of implementing this switching in this function, what if `loadCollaborators` only returned the map of collaborators and the callers were responsible for checking and initializing the cache?

That would also let you replace the `direct` parameter and take the `affiliation` value directly.

Reply:
I felt like it was a little more clunky if the clients need to worry about the caching, but I'm open to changing it if you feel like this is a worse outcome.